### ISSUE TITLE
方向分类器训练出现的问题

### ISSUE THREAD

**Issue by wu-yakun** [PaddlePaddle/PaddleOCR#3438]
方向分类器的训练txt文件如下图所示：
![3ebff81e36b04fd68a9c61ed45f2921](https://user-images.githubusercontent.com/48200202/127111664-ea2b6732-11d6-441f-9d82-c9edeff60f07.png)

问题：训练txt文件中加上标签是180度的图片，训练就报错，报错内容如下图所示：
![2d087b5ae1d1e0569a4a23df395ed79](https://user-images.githubusercontent.com/48200202/127111892-8220496d-9a87-4009-9a4c-e21ece8b9c7f.png)
错误意思是：调用Python对象时超出了最大递归深度，致命的Python错误：无法从堆栈溢出中恢复。

若是把标签是180度的图片去除，训练就正常。此时的训练txt文件如下图所示：
![3b6b8310dabba7166c9e7010a85a163](https://user-images.githubusercontent.com/48200202/127112262-27b09ec2-19a1-45a4-99c4-a33aff990d3b.png)

这个问题好奇怪，请小哥哥小姐姐指导下，不胜感激。

**Comment by WenmuZhou** [comment]
你的config贴上来看看

**Comment by wu-yakun** [comment]
`Global:
  use_gpu: true
  epoch_num: 100
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./trained_model/cls/
  save_epoch_step: 120
  # evaluation is run every 5000 iterations after the 4000th iteration
  eval_batch_step: [0, 1000]
  cal_metric_during_train: True
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_words_en/word_10.png
  label_list: ['0','180']

Architecture:
  model_type: cls
  algorithm: CLS
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.35
    model_name: small
  Neck:
  Head:
    name: ClsHead
    class_dim: 2

Loss:
  name: ClsLoss

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: ClsPostProcess

Metric:
  name: ClsMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/character_cls/
    label_file_list:
      - ./train_data/character_cls/train.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - ClsLabelEncode: # Class handling label
      - RecAug:
          use_tia: False
      - RandAugment:
      - ClsResizeImg:
          image_shape: [3, 48, 192]
      - KeepKeys:
          keep_keys: ['image', 'label'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 8
    drop_last: True
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/character_cls/
    label_file_list:
      - ./train_data/character_cls/train.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - ClsLabelEncode: # Class handling label
      - ClsResizeImg:
          image_shape: [3, 48, 192]
      - KeepKeys:
          keep_keys: ['image', 'label'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 8
    num_workers: 4
`

**Comment by wu-yakun** [comment]
`Global:
  use_gpu: true
  epoch_num: 100
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./trained_model/cls/
  save_epoch_step: 120
  # evaluation is run every 5000 iterations after the 4000th iteration
  eval_batch_step: [0, 1000]
  cal_metric_during_train: True
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_words_en/word_10.png
  label_list: ['0','180']

Architecture:
  model_type: cls
  algorithm: CLS
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.35
    model_name: small
  Neck:
  Head:
    name: ClsHead
    class_dim: 2

Loss:
  name: ClsLoss

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: ClsPostProcess

Metric:
  name: ClsMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/character_cls/
    label_file_list:
      - ./train_data/character_cls/train.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - ClsLabelEncode: # Class handling label
      - RecAug:
          use_tia: False
      - RandAugment:
      - ClsResizeImg:
          image_shape: [3, 48, 192]
      - KeepKeys:
          keep_keys: ['image', 'label'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 8
    drop_last: True
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/character_cls/
    label_file_list:
      - ./train_data/character_cls/train.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - ClsLabelEncode: # Class handling label
      - ClsResizeImg:
          image_shape: [3, 48, 192]
      - KeepKeys:
          keep_keys: ['image', 'label'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 8
    num_workers: 4
`

**Comment by wu-yakun** [comment]
配置文件如上所示。

**Comment by WenmuZhou** [comment]
看起来是解析180的时候报错了，然后就循环解析去了，你在simpledataset的getitem函数里debug一下看看哪里报错了

**Comment by wu-yakun** [comment]
感谢大佬，解决了，问题就出在getitem函数中！

**Comment by WenmuZhou** [comment]
能描述一下是啥错误吗

**Comment by Jackzhangfei** [comment]
请问一下是如何解决的呢

**Comment by aaaaalun** [comment]
直接修改代码，让角度的字符串里面没有\r即可。

**Comment by LWQ2EDU** [comment]
> 直接修改代码，让角度的字符串里面没有\r即可。

把标注文件中的换行符都换成\r解决了
